### PR TITLE
feat(icons): make it easier to replace default entity icons

### PR DESCRIPTION
**Pull request by hypeJunction**
Entity icons now reside in a predictable location in the view system.

BREAKING CHANGE
Moves default entity icon files and remaps them to match the new
icon fallback logic.
Removes some legacy group and user icon handling

### CODE REVIEW

**Comment by hypeJunction** [comment]
I guess this would be time to switch to svg icons... #8019

**Review by jdalsem — APPROVED** [review]

**Comment by mrclay** [comment]
> mod/groups/graphics/defaultlarge.gif → views/default/icon/group/default/large.gif

We're OK with groups icons placed back into core views?

**Review by mrclay — APPROVED** [review]

**Comment by hypeJunction** [comment]
> We're OK with groups icons placed back into core views?

For the sake of convenience.

**Comment by jdalsem** [comment]
no

**Comment by hypeJunction** [comment]
You are not ok with it? I don't have a strong opinion. Can just move them back into groups plugin.

**Comment by jdalsem** [comment]
keep them with the plugin please

**Comment by mrclay** [comment]
Move back into groups, rebase, and should we get one more opinion?

**Comment by juho-jaakkola** [comment]
My vote is on keeping them with the plugin.

**Comment by jdalsem** [comment]
great lets finish this up then

File: mod/groups/actions/groups/edit.php [hunk 1]
```diff
@@ -210,21 +210,7 @@
 	));
 }
 
-$has_uploaded_icon = (!empty($_FILES['icon']['type']) && substr_count($_FILES['icon']['type'], 'image/'));
-
-if ($has_uploaded_icon) {
-	$filehandler = new ElggFile();
-	$filehandler->owner_guid = $group->owner_guid;
```
Comment (jdalsem) [inline mod/groups/actions/groups/edit.php]: i believe this breaks already uploaded images... that is why https://github.com/Elgg/Elgg/issues/4683 exists... fine by me to merge this now... upgrade can fix the old locations.

Comment (hypeJunction) [inline mod/groups/actions/groups/edit.php]: I would have left a comment if it were so. The icon file hook takes care of this logic. I think this was some interim step in the process.

Comment (jdalsem) [inline mod/groups/actions/groups/edit.php]: ah ok... i did look if it was part of this PR, so it was probably taken care of earlier

**Comment by hypeJunction** [comment]
Group icons are back in the groups plugin, rebased.

**Pull request merged**